src/Core/Form/IdentifiableObject/DataProvider/CurrencyFormDataProvider.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataProvider;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Currency\DTO\CurrencyForFormEditing;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Query\GetCurrencyForFormEditing;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+
+/**
+ * Class CurrencyFormDataProvider
+ */
+final class CurrencyFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var array
+     */
+    private $contextShopIds;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @param CommandBusInterface $queryBus
+     * @param array $contextShopIds
+     */
+    public function __construct(CommandBusInterface $queryBus, array $contextShopIds)
+    {
+        $this->contextShopIds = $contextShopIds;
+        $this->queryBus = $queryBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData($id)
+    {
+        /** @var CurrencyForFormEditing $result */
+        $result = $this->queryBus->handle(new GetCurrencyForFormEditing(new CurrencyId((int) $id)));
+
+        return [
+            'iso_code' => $result->getIsoCode(),
+            'exchange_rate' => $result->getExchangeRate(),
+            'shop_association' => $result->getAssociatedShopIds(),
+            'active' => $result->isEnabled(),
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultData()
+    {
+        return [
+            'exchange_rate' => 1,
```

Comment:
what is 1? how about storing is as constant in `ExchangeRate` VO as `ExchangeRate::DEFAULT_RATE`?